docs(extmarks): Improve documentation

muniter opened the pull request:
Attempts to resolve #11456 by providing a more clear example.
Also fixes mistakes in the example providing incorrect parameter count to the API functions.

muniter committed:
docs: extmarks indexing #15311

fix #11456

vigoux commented on runtime/doc/api.txt:
```diff
@@ -463,7 +477,15 @@ We can get all marks in a buffer for our namespace (or by a range): >
+    0 e|ample..
+       ^ This is the extmark position
+
+    Note: An extmark "gravity" meaning the direction it will be shifted after a
+    text edit can be adjusted see |nvim_buf_set_extmark()| for more information.
```
This sounds a bit weird, maybe:
```suggestion
    Note: An extmark "gravity" is the direction it will be shifted after a
    text edit. See |nvim_buf_set_extmark()| for more information.
```

seandewar commented on runtime/doc/api.txt:
```diff
@@ -463,7 +477,15 @@ We can get all marks in a buffer for our namespace (or by a range): >
 
 Deleting all text surrounding an extmark does not remove the extmark. To
-remove an extmark use |nvim_buf_del_extmark()|.
+remove an extmark use |nvim_buf_del_extmark()|. Deleting "x" in our example
+would result in the following.
+
+      0 12345678
+    0 e|ample..
+       ^ This is the extmark position
+
+    Note: An extmark "gravity" meaning the direction it will be shifted after a
```
Maybe fence this like your other example was:

```suggestion
remove an extmark use |nvim_buf_del_extmark()|. Deleting "x" in our example
would result in the following: <

      0 12345678
    0 e|ample..
       ^ This is the extmark position
>
    Note: An extmark "gravity" meaning the direction it will be shifted after a
```

muniter commented:
Have amended both suggestions, thanks! I wasn't that familiar with code blocks. Now I've improved it so that the `^ This is the extmark position` is out the code block and the highlight makes it more clear.

runiq committed:
docs: extmark indexing #12742

Extmarks mostly use api-indexing, except for nvim_buf_get_extmarks(),
which uses api-indexing with inclusive ranges.

ref #11456

justinmk committed:
docs: extmarks indexing #15311

ref #11456

justinmk commented:
pushed some changes, also included https://github.com/neovim/neovim/pull/12742 

thanks for giving attention to this.
